debezium-core/src/main/java/io/debezium/pipeline/ChangeEventSourceCoordinator.java

```diff
@@ -311,15 +311,15 @@ protected CatchUpStreamingResult executeCatchUpStreaming(ChangeEventSourceContex
     }
 
     protected void streamEvents(ChangeEventSourceContext context, P partition, O offsetContext) throws InterruptedException {
-        initStreamEvents(partition, offsetContext);
-        getSignalProcessor(previousOffsets).ifPresent(signalProcessor -> registerSignalActionsAndStartProcessor(signalProcessor,
-                eventDispatcher, this, connectorConfig));
-
         if (snapshotterService != null && !snapshotterService.getSnapshotter().shouldStream()) {
             LOGGER.info("Streaming is disabled for snapshot mode {}", snapshotterService.getSnapshotter().name());
             return;
         }
 
+        initStreamEvents(partition, offsetContext);
+        getSignalProcessor(previousOffsets).ifPresent(signalProcessor -> registerSignalActionsAndStartProcessor(signalProcessor,
```

Comment:
If we move this here, it will not be possible to execute a `blocking` snapshot.

Reply:
I don't understand @mfvitale, why would blocking snapshots be possible without streaming?

Reply:
Yes, you're right but maybe in future blocking snapshot, since it is just an on-demand initial snapshot, could be possible to be trigger even if streaming is not running.  

Also, we are excluding the possibility to execute any signal.